tests/test_scalars.py

```diff
@@ -0,0 +1,713 @@
+    check(assert_type(p.qyear, int), int)
+    check(assert_type(p.second, int), int)
+    check(assert_type(p.start_time, pd.Timestamp), pd.Timestamp)
+    check(assert_type(p.week, int), int)
+    check(assert_type(p.weekday, int), int)
+    check(assert_type(p.weekofyear, int), int)
+    check(assert_type(p.year, int), int)
+    check(assert_type(p.freq, BaseOffset), Day)
+    check(assert_type(p.ordinal, int), int)
+
+    p2 = pd.Period("2012-1-1", freq="2D")
+    check(assert_type(p2.freq, BaseOffset), Day)
+
+    as0 = pd.Timedelta(1, "D")
+    as1 = dt.timedelta(days=1)
+    as2 = np.timedelta64(1, "D")
+    as3 = np.int64(1)
+    as4 = int(1)
+    as5 = pd.period_range("2012-1-1", periods=10, freq="D")
+    as6 = pd.Period("2012-1-1", freq="D")
+
+    check(assert_type(p + as0, pd.Period), pd.Period)
+    check(assert_type(p + as1, pd.Period), pd.Period)
+    check(assert_type(p + as2, pd.Period), pd.Period)
+    check(assert_type(p + as3, pd.Period), pd.Period)
+    check(assert_type(p + as4, pd.Period), pd.Period)
+
+    check(assert_type(p - as0, pd.Period), pd.Period)
+    check(assert_type(p - as1, pd.Period), pd.Period)
+    check(assert_type(p - as2, pd.Period), pd.Period)
+    check(assert_type(p - as3, pd.Period), pd.Period)
+    check(assert_type(p - as4, pd.Period), pd.Period)
+    check(assert_type(p - as5, pd.Index), pd.Index)
+    check(assert_type(p - as6, BaseOffset), Day)
+
+    check(assert_type(as0 + p, pd.Period), pd.Period)
+    check(assert_type(as1 + p, pd.Period), pd.Period)
+    check(assert_type(as2 + p, pd.Period), pd.Period)
+    check(assert_type(as3 + p, pd.Period), pd.Period)
+    check(assert_type(as4 + p, pd.Period), pd.Period)
+    # TOOD: PeriodIndex should have a __sub__ with correct types, this op is valid
+    #  and so the assert_type is skipped
+    check(as5 - p, pd.Index)  # type: ignore[operator]
+
+    check(assert_type(p.__radd__(as0), pd.Period), pd.Period)
+    check(assert_type(p.__radd__(as1), pd.Period), pd.Period)
+    check(assert_type(p.__radd__(as2), pd.Period), pd.Period)
+    check(assert_type(p.__radd__(as3), pd.Period), pd.Period)
+    check(assert_type(p.__radd__(as4), pd.Period), pd.Period)
+
+    p - p
+
+    c0 = pd.Timestamp("2012-1-1")
+    c1 = dt.datetime(2012, 1, 1)
+    c2 = dt.date(2012, 1, 1)
+    c3 = np.datetime64(1, "ns")
+    c4 = 0
+    c5 = np.int64(0)
+    c6 = pd.Period("2012-1-1", freq="D")
+    c7 = pd.period_range("2012-1-1", periods=10, freq="D")
+    c8 = pd.date_range("2012-1-1", periods=10, freq="D")
+
+    check(assert_type(p == c0, bool), bool)
+    check(assert_type(p == c1, bool), bool)
+    check(assert_type(p == c2, bool), bool)
+    check(assert_type(p == c3, bool), bool)
+    check(assert_type(p == c4, bool), bool)
+    check(assert_type(p == c5, bool), bool)
+    check(assert_type(p == c7, npt.NDArray[np.bool_]), np.ndarray)
+    check(assert_type(p == c8, npt.NDArray[np.bool_]), np.ndarray)
+
+    check(assert_type(p != c0, bool), bool)
```

Comment:
I kind fo feel like we should not be adding this test here.  This a test of pandas correctness not of type.  And TBH, when writing lots of tests, it is helpful to not have to worry about values and just types, e.g., using np.array([2192190], dtype="M8[ns]") - I don't know or really care what the value of this is here. 

I think if one wanted to go down this route, tests would need to ahve both `True` and `False` for all ops to ensure that the op is valid, esp for `==` and `!=` since these often pass on anything even if not really comparable.